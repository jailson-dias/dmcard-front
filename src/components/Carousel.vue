<template>
  <v-carousel
    :show-arrows="false"
    :height="height"
    cycle
    hide-delimiter-background
    :interval="3000"
  >
    <v-carousel-item v-for="(item,i) in items" :key="i">
      <v-img :src="item.src" aspect-ratio="1.7" :contain="item.contain" height="100%" width="100%"></v-img>
    </v-carousel-item>
  </v-carousel>
</template>

<script>
export default {
  data() {
    return {
      items: [
        {
          src: require("@/assets/app-mockup.png"),
          contain: true
        },
        {
          src: require("@/assets/atuamos.jpg")
        },
        {
          src: require("@/assets/perguntas.jpg")
        }
      ],
      height: 0
    };
  },
  mounted() {
    this.onResize();
  },

  methods: {
    onResize() {
      this.height = window.innerHeight;
    }
  }
};
</script>

<style>
.v-carousel__controls {
  justify-content: flex-start !important;
}

.v-carousel__controls__item {
  height: 15px !important;
  width: 15px !important;
  margin-left: 1px !important;
  margin-right: 1px !important;
}

.v-carousel__controls__item .v-btn__content .v-icon {
  font-size: 10px !important;
  height: 15px !important;
  width: 15px !important;
}

.v-icon {
  color: #00000055 !important;
}
</style>