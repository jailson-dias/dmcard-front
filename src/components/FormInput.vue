<template>
  <div id="request-card-form">
    <div class="d-flex flex-column">
      <h2
        class="d-flex text-center display-1 font-weight-bold"
        :style="{fontSize: titleSize + 'px'}"
      >{{title}}</h2>
      <p class="text-justify" :style="{fontSize: subtitleSize + 'px'}">{{subtitle}}</p>

      <v-form v-model="valid" ref="form">
        <Input
          v-for="(input, i) in inputs"
          :label="input.label"
          :key="'label' + i"
          @input="event => input.set(event)"
          :rules="input.rules"
          :mask="input.mask"
          :money="input.money"
          :prefix="input.prefix"
          :type="input.type"
          :defaultValue="input.defaultValue"
        />
        <Button :text="button.text" @click.native="button.submit(valid)" />
      </v-form>
    </div>
  </div>
</template>

<script>
import Input from "./Input";
import Button from "./Button";

export default {
  props: {
    title: String,
    subtitle: String,
    titleSize: [String, Number],
    subtitleSize: [String, Number],
    inputs: {
      validator: value => {
        if (Array.isArray(value)) {
          if (!value[0] || !value[0].label) {
            return false;
          }
          return true;
        }
        return false;
      },
      required: true
    },
    button: {
      validator: value => {
        if (value && value.text) {
          return true;
        }
        return false;
      },
      required: true
    }
  },
  data() {
    return {
      valid: false,
      windowSize: 100
    };
  },
  components: {
    Input,
    Button
  }
};
</script>

<style>
#request-card-form {
  padding-left: 60px !important;
  padding-right: 60px !important;
}

#request-card-form > div > h2 {
  margin-top: 70px;
}

#request-card-form > div > p {
  font-size: 20px;
  margin-top: 20px;
  margin-bottom: 20px;
}
</style>