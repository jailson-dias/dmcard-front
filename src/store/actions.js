import api from "../api";

export default {
  reAuthenticateUser({ commit }, { router, isDashboard, isLogin }) {
    if (localStorage.user) {
      try {
        let user = JSON.parse(localStorage.user);
        commit("setUser", user);
        api.setCredentials(`Bearer ${user.token}`);
        if (!isDashboard) {
          router.push("/dashboard");
        }
      } catch (e) {
        if (!isLogin) {
          router.push("/login");
        }
      }
    } else if (!isLogin) {
      router.push("/login");
    }
  },
  login({ commit }, { router, username, password }) {
    return api.user.login({ username, password }).then(user => {
      commit("setUser", user);
      localStorage.user = JSON.stringify(user);
      api.setCredentials(`Bearer ${user.token}`);
      router.push("/dashboard");
      return user;
    });
  },
  logout({ commit }, { router, isLogin }) {
    api.setCredentials("");
    localStorage.removeItem("user");
    commit("setUser", {});
    if (!isLogin) {
      router.push("/login");
    }
  },
  createRequestCreditCard({ commit }, { name, cpf, email, phone, income }) {
    return api.requestCreditCard
      .create({ name, cpf, email, phone, income })
      .then(request => {
        commit("setCurrentRequest", request);
        return request;
      });
  },
  listAllRequestCreditCard({ commit }) {
    return api.requestCreditCard.listAll().then(requests => {
      commit("setRequestCreditCard", requests);
      commit("setRequestCreditCardScreenTitle", "Todas as solicitações");
    });
  },
  listAcceptedRequestCreditCard({ commit }) {
    return api.requestCreditCard.listAccepted().then(requests => {
      commit("setRequestCreditCard", requests);
      commit("setRequestCreditCardScreenTitle", "Solicitações aceitas");
    });
  },
  listRefusedRequestCreditCard({ commit }) {
    return api.requestCreditCard.listRefused().then(requests => {
      commit("setRequestCreditCard", requests);
      commit("setRequestCreditCardScreenTitle", "Solicitações recusadas");
    });
  },
  removeRequestCreditCard({ commit }, { id }) {
    return api.requestCreditCard.delete({ id }).then(request => {
      commit("removeRequestCreditCard", request);
    });
  }
};
