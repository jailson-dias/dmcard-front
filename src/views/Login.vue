<template>
  <v-container class="pa-0 mx-0">
    <v-row no-gutters>
      <v-col :cols="8">
        <Carousel />
      </v-col>
      <v-col :cols="4" class="d-flex flex-column align-center">
        <img src="@/assets/nav-dmcard.png" alt="logo" class="logo-dmcard" />
        <FormInput
          subtitle="Usuário: admin, senha: admin"
          subtitleSize="15"
          :inputs="inputs"
          :button="button"
        />
      </v-col>
      <RequestCreditCardDialog
        title="Acesso negado"
        text="Usuário e/ou senha incorretos"
        :dialog="modal"
        @set="hideModal"
      />
    </v-row>
  </v-container>
</template>

<script>
import Carousel from "../components/Carousel";
import FormInput from "../components/FormInput";
import RequestCreditCardDialog from "../components/RequestCreditCardDialog";

import { mapState, mapActions } from "vuex";

const user = {
  username: "admin",
  password: "admin"
};
export default {
  components: {
    Carousel,
    FormInput,
    RequestCreditCardDialog
  },
  data() {
    return {
      modal: false,
      user,
      inputs: [
        {
          label: "Usuário",
          set: this.setUsername,
          defaultValue: user.username
        },
        {
          label: "Senha",
          type: "password",
          defaultValue: user.password,
          set: this.setPassword
        }
      ],
      button: {
        text: "Entrar",
        submit: this.submit
      }
    };
  },
  computed: {
    ...mapState({
      username: state => state.user.username
    })
  },
  created() {
    this.reAuthenticateUser({ router: this.$router, isLogin: true });
  },
  methods: {
    ...mapActions(["login", "reAuthenticateUser"]),
    setUsername(value) {
      this.user.username = value;
    },
    setPassword(value) {
      this.user.password = value;
    },
    submit() {
      this.login({ router: this.$router, ...this.user }).catch(err => {
        this.modal = true;
      });
    },
    hideModal() {
      this.modal = false;
    }
  }
};
</script>

<style scoped>
.logo-dmcard {
  height: 70px;
  width: 70px;
  margin-top: 120px;
  margin-bottom: -80px;
}
</style>