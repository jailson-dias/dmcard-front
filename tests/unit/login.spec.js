import { shallowMount, createLocalVue } from "@vue/test-utils";
import Login from "@/views/Login.vue";
import Carousel from "@/components/Carousel.vue";
import FormInput from "@/components/FormInput.vue";
import Router from "vue-router";
import Vue from "vue";
import Vuex from "vuex";
import actions from "@/store/actions";
import initialState from "@/store/state";
import Vuetify from "vuetify";

const localVue = createLocalVue();
localVue.use(Router);
localVue.use(Vuex);
Vue.use(Vuetify);

describe("Login page", () => {
  let vuetify;
  let state;
  const build = () => {
    const wrapper = shallowMount(Login, {
      localVue,
      vuetify,
      store: new Vuex.Store({
        state,
        actions
      })
    });

    return {
      wrapper,
      Carousel: () => wrapper.find(Carousel),
      FormInput: () => wrapper.find(FormInput)
    };
  };

  beforeEach(() => {
    jest.resetAllMocks();
    vuetify = new Vuetify();
    state = { ...initialState };
  });

  it("É uma página do vue", () => {
    const { wrapper } = build();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("A página tem o carousel e o form para preencher os dados", () => {
    const { Carousel, FormInput } = build();

    expect(Carousel().exists()).toBe(true);
    expect(FormInput().exists()).toBe(true);
  });
});
