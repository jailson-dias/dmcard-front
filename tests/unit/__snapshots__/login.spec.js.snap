// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page É uma página do vue 1`] = `
<v-container-stub tag="div" class="pa-0 mx-0">
  <v-row-stub tag="div" nogutters="true">
    <v-col-stub cols="8" tag="div">
      <carousel-stub></carousel-stub>
    </v-col-stub>
    <v-col-stub cols="4" tag="div" class="d-flex flex-column align-center"><img src="@/assets/nav-dmcard.png" alt="logo" class="logo-dmcard">
      <forminput-stub subtitle="Usuário: admin, senha: admin" subtitlesize="15" inputs="[object Object],[object Object]" button="[object Object]"></forminput-stub>
    </v-col-stub>
  </v-row-stub>
</v-container-stub>
`;
